### ISSUE TITLE
MyLocationPlugin configurable location tracking

### ISSUE THREAD

**Issue by kuosman** [oskariorg/oskari-docs#126]
# Use case
* user come mobile mode to map
* automatically zoom map to current location and show tracking ball and accuracy circle
* user can zoom and move map, map is not centered back user location when user move around, only tarcking ball / accurary circle location move

# Development
* add new configuration and handling in MyLocationPlugin: `{track: true|false}` (defaults false) or if want seperate mobile/desktop configuration then use: `{track: {mobile: true|false, desktop: true|false}}` (defaults also false if not found)
* if tracking is configured to false, then nothing change
* if tracking is configured to true:
    * first time coming map zoom and locate user from map
    * show tracking ball and accuracy circle when user location changed
    * set icon to active
    * user can toggle tracking state by clickin icon

**Comment by ZakarFin** [comment]
This is only available for db config? Should there be a selection under the user location in publisher for the mode? Like track movement vs. single use. And another if automatic map centering/tracking on location should be done at startup?

Should we just change the functionality to always be tracking and if you just want to center the location you could switch it off tracking after clicking it on?

**Comment by kuosman** [comment]
Hmm, ok, maybe selections need also in publisher.
The customer user case is at some view need to show tracking location when user is only mobile mode, this configuration/selection not needed be in publisher, or maybe some selection also for this (Use in etc) ?

# Proposal

## Publisher
Under the user location:
* Mode: 
  * single use (functionality like now, selected by default)
  * track movement  (show user location ball and accurary circle, not actually move map to user location)
* Center map automatically user location when coming to map

(

* Use in:
  * All map sizes
  * Desktop map sizes
  * Mobile map sizes

)

```
{
    mode: 'single' || 'track',
    centerMapAutomatically: false | true,
    use: 'all' | 'desktop' | 'mobile'
}
```

## DB config

Uses same config syntax than Publisher.

## Functionality

If track movement is configured:
User can stop tracking by pressing icon when map zooms to user location and stop tracking.

## Default configuration

Default configuration is same than current usage, like this config:  `{mode: 'single', centerMapAutomatically: false, use: 'all'}` or is better to always show user tracking location ?

**Comment by ZakarFin** [comment]
The centerMapAutomatically config with some server-side magic to toggle it programmatically if the has for example coordinates would make this plugin obsolete https://github.com/oskariorg/oskari-frontend/blob/1.53.0/bundles/mapping/mapmodule/plugin/location/GeoLocationPlugin.js. Would be good to have just one plugin for this. This is the related server-side logic: https://github.com/oskariorg/oskari-server/blob/1.53.0/control-base/src/main/java/fi/nls/oskari/control/view/modifier/bundle/MapfullHandler.java#L147-L150

I'm not sure about the `use` config. So you would want to show/hide the location plugin based on the browser _size_ being small (mobile) or large (desktop)?

**Comment by ZakarFin** [comment]
Why wouldn't you want to move the map when tracking? Considering the user location might not be in the map viewport.

**Comment by ZakarFin** [comment]
Just read the earlier specs again. So track doesn't actually draw the route in your proposal it just continues to update the user location without moving the map? So I guess better names for the modes could be:
- single (only centers map once when the button is clicked, moves map to location)
- continuous (Shows the accuracy circle and user location is updated periodically, toggled on/off by button, moves map only if the user location is outside of the viewport) 

I'm thinking "track" mode would draw the route of the user and maybe also center the map on user location whenever moving but this is something that can be left out and implemented at a later date if it's not something you need right now?

Maybe you could squeeze in the refactoring where GeoLocationPlugin.js would be dropped and MyLocationPlugin with the server side logic on centerMapAutomatically would be used instead?

**Comment by ZakarFin** [comment]
Had a quick online meeting about this and we are moving forward with the plan above. The `use` key will be renamed to something more describing like showOnlyOn: 'mobile' | 'desktop'.

**Comment by kuosman** [comment]
Also have quick specs again and showOnlyOn parameter is renamed to mobileOnly: true | false

**Comment by ZakarFin** [comment]
Implemented in oskariorg/oskari-frontend#979
